mitodl/open-discussions
# Copy mitodl/course_catalog app into discussions

#### Pre-Flight checklist

#### What are the relevant tickets?
Partially addresses #1720
Closes #1774

#### What's this PR do?
Copies the `course_catalog` app from the mitodl/course-catalog repo into discussions.

#### How should this be manually tested?
- Set the following env variables (I can send you the values):
  ```
  EDX_API_URL=https://api.edx.org/catalog/v1/catalogs/10/courses
  EDX_API_CLIENT_ID=<value>
  EDX_API_CLIENT_SECRET=<value>

  OCW_CONTENT_BUCKET_NAME=ocw-content-storage
  OCW_CONTENT_ACCESS_KEY=<value>
  OCW_CONTENT_SECRET_ACCESS_KEY=<value>
  ```
- To upload OCW thumbnail images the following would need to be set, devops will be creating them 
  shortly or you can use a personal bucket for testing, or just set uploading to False.
  ```
  OCW_LEARNING_COURSE_BUCKET_NAME=<value>
  OCW_LEARNING_COURSE_ACCESS_KEY=<value>
  OCW_LEARNING_COURSE_SECRET_ACCESS_KEY=<value>
  ```
- In a shell, run `from course_catalog.tasks import *;get_edx_data()`
- After it completes there should be 672 `Course` objects (with `platform`=`mitx`), 31 `CourseTopic`s, 440 `CoursePrice`s, and 2225 `CourseInstructor`s.
- Run `get_ocw_data(upload_to_s3=False)`.  Wait until a few are processed then interrupt the command and verify that a few `Course`s with `platform`=`ocw` exist.

- Try out the following URL's and make sure they work:
  - http://od.odl.local:8063/api/v0/ocw-course-report
  - http://od.odl.local:8063/api/v0/courses/
  - http://od.odl.local:8063/api/v0/courses/new/
  - http://od.odl.local:8063/api/v0/courses/featured/ (empty list unless you manually set a course to featured)
